src/components/partnership/partnership.service.ts

```diff
@@ -369,6 +382,11 @@ export class PartnershipService {
       }
     }
 
+    if (input.primary) {
+      const projectId = await this.getProjectIdByPartnershipId(input.id);
+      await this.setPrimaryForOldPartnerships(projectId, false);
+    }
```

Comment:
What do you think about a `movePrimaryPartnership` method instead? That could work for create, update, and delete.

logically,
- match current primary partnership, its project, and all other partnerships
- set current to primary false
- set first of other partnerships to primary true

I think that could consolidate logic and reduce queries.